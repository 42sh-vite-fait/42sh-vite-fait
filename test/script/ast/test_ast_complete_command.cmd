echo 'useless commands'; first=$(ls -1 2>/dev/null | head -n 1); test -d ${first} && (touch ${first}/file && cat /dev/random | base64 | head -c 42) 2>&1 >output || mkdir ${first}
