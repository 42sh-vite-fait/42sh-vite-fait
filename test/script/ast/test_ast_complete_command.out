AST:
LIST: ;
	LIST: ;
		SIMPLE_COMMAND: echo
			args#1: 'useless commands'
		SIMPLE_COMMAND: first=$(ls -1 2>/dev/null | head -n 1)
	AND_OR: ||
		AND_OR: &&
			SIMPLE_COMMAND: test
				args#1: -d
				args#2: ${first}
			SUBSHELL: ()
				redirections#0: IO=2, Operator=GREATAND, Word=1
				redirections#1: IO=1, Operator=GREAT, Word=output
				AND_OR: &&
					SIMPLE_COMMAND: touch
						args#1: ${first}/file
					PIPE_SEQUENCE: |
						PIPE_SEQUENCE: |
							SIMPLE_COMMAND: cat
								args#1: /dev/random
							SIMPLE_COMMAND: base64
						SIMPLE_COMMAND: head
							args#1: -c
							args#2: 42
		SIMPLE_COMMAND: mkdir
			args#1: ${first}
